nn-git-log/README.md

```diff
@@ -0,0 +1,113 @@
+For example information like when the commit was made or who made it is not really important when trying to look at the bigger picture.  
+This is also not practical when looking at projects with 10s or 100s of commits, as we would be lost in the information
+
+### Flag oneline
+
+Luckily there are special flags we can use to format the output.  
+If we add the flag, `--oneline`, git will shorten each commit to oneline, only preserving the important information.  
+Looking at the man page for this command:  
+```bash
+Documentation:
+--oneline
+This is a shorthand for "--pretty=oneline --abbrev-commit" used together.
+--abbrev-commit
+Instead of showing the full 40-byte hexadecimal commit object name, show only a partial prefix. Non default number of digits can be specified with "--abbrev=<n>" (which also modifies diff output, if it is displayed).
+```
+Thus when running the command.
+```bash
+git log --oneline
+```
+![Git log with only oneline](res/small_picture_2.png)
+
+And boom, this was what we wanted at the start. Now we can see that we have added the landing page and also changed the extension of the landing page.  
+
+But there is something weird here, we can also see that the `origin/master` commit together with the commits of our branch. This is because as we know, branches have to start off(be based off) somewhere. As such, we can make a pretty good guess that this `frontend/landing` was made off the master branch.  
+
+Luckily, we don't have to guess as there is a flag which will show us the relationship between these commits.  
+
+### Flag graph
+To make git log show the relationships between commits, we just have to add the `--graph` flag. This flag shows a 'graph' of the commits, which for us means the relationship the commits have.  
+Combining this with `--oneline` flag we showed above, makes for a very clean and concise git log.  
+
+(I will leave it to the reader to try out what `--graph` flag does on its own)
+
+```bash
+git log --oneline --graph
+```
+![Git log with oneline and graph](res/small_picture_3.png)
+
+## Looking at the big picture
+Now that you know how to look at commits tagged to your branch, you might also want to see how the entire repository might look like.  
+Luckily we have a flag to look at the log of the entire repository.  
+
+### Flag all
+Using the `--all` flag, this will show the git log, including all the commits from every branch you might have in the respository.  
+We will also be combining this flag with all previous flags to get the best overview of a project. (In my opnion)
+```bash
+git log --oneline --graph --all
+```
+![Git log with oneline graph and all](res/big_picture_1.png)
+
+Now we can see that our frontend landing page feature has already been merged down to master. We can also see that after git fetch(To fetch any changes from github), our teammate B had already made more commits on his feature and our local `backend/start-server` branch is behind.  
```

Comment:
Perhaps discuss how the lines on the left hand side of `git log` show the merge history as well? So that this doesn't suddenly introduce a new concept of having the frontend feature branch be merged with the `master` branch. Also, it might be good to specify that the merge has occurred in the local `master` branch and not the remote. I will be going into the naming convention of branching too so that this isn't a new bombshell of knowledge on them

Reply:
> Perhaps discuss how the lines on the left hand side of `git log` show the merge history as well? So that this doesn't suddenly introduce a new concept of having the frontend feature branch be merged with the `master` branch.

Agreed.
Initially, I was thinking of adding a section on how to read and relate the left-hand side to the conceptual diagrams of git branching but thought that was too much. I'm still on the french about this.

> Also, it might be good to specify that the merge has occurred in the local `master` branch and not the remote.

Will add more explanations to make it more explicit.

> I will be going into the naming convention of branching too so that this isn't a new bombshell of knowledge on them

Not sure if I followed the naming convention on this haha XD.

Reply:
> Initially, I was thinking of adding a section on how to read and relate the left-hand side to the conceptual diagrams of git branching but thought that was too much. I'm still on the french about this.

I think you can connect this portion with the image of the GitHub visualisation tool. GitHub's tool is pretty clear to understand.

> Not sure if I followed the naming convention on this haha XD.

I believe the convention on GitHub is just `<username>:<branch>`

Reply:
Resolving this for now but will bring this up again in the new review :+1: